lib/view.dart

```diff
@@ -156,49 +156,109 @@ void renderChartOptions(bool isComparisonChecked, bool isNormalisationChecked) {
-    ..classes = ['form-check-label']
-    ..innerText = 'Compare data';
-  comparisonWrapper.append(comparisonOption);
-  comparisonWrapper.append(comparisonLabel);
-  optionsCol.append(comparisonWrapper);
+    checkboxCol.append(checkboxWithLabel);
 
-  var normalisationWrapper = html.DivElement()
-    ..classes = ['form-check', 'form-check-inline'];
-  var normalisationOption = html.InputElement()
-    ..type = 'checkbox'
-    ..id = 'data-normalisation-option'
-    ..classes = ['form-check-input']
-    ..checked = isNormalisationChecked
-    ..onChange.listen((e) {
-      command(UIAction.toggleDataNormalisation,
-          ToggleOptionEnabledData((e.target as html.CheckboxInputElement).checked));
+    var filterDropdown =
+        _getDropdown(filterOptions[key].toList(), '__all', (String value) {
+      command(UIAction.setFilterValue, SetFilterValueData(key, value));
     });
-  var normalisationLabel = html.LabelElement()
-    ..htmlFor = 'data-normalisation-option'
-    ..classes = ['form-check-label']
-    ..innerText = 'Normalise data';
-  normalisationWrapper.append(normalisationOption);
-  normalisationWrapper.append(normalisationLabel);
-  optionsCol.append(normalisationWrapper);
+    filterCol.append(filterDropdown);
+
+    filterRow.append(checkboxCol);
+    filterRow.append(filterCol);
+    optionsCol.append(filterRow);
+
+    if (!shouldRenderComparisonFilters) continue;
+    var comparisonFilterCol = html.DivElement()..classes = ['col-3'];
+    var comparisonFilterDropdown =
+        _getDropdown(filterOptions[key].toList(), '__all', (String value) {
+      command(
+          UIAction.setComparisonFilterValue, SetFilterValueData(key, value));
+    });
+    comparisonFilterCol.append(comparisonFilterDropdown);
+    filterRow.append(comparisonFilterCol);
+  }
 
   wrapper.append(labelCol);
   wrapper.append(optionsCol);
   content.append(wrapper);
 }
 
+html.DivElement _getCheckboxWithLabel(
+    String id, String label, bool isChecked, Function(bool) onChange) {
+  var checkboxWrapper = html.DivElement()
+    ..classes = ['form-check', 'form-check-inline'];
+  var checkboxOption = html.InputElement()
+    ..type = 'checkbox'
+    ..id = id
+    ..classes = ['form-check-input']
+    ..checked = isChecked
+    ..onChange.listen(
+        (e) => onChange((e.target as html.CheckboxInputElement).checked));
+  var checkboxLabel = html.LabelElement()
+    ..htmlFor = id
+    ..classes = ['form-check-label']
+    ..innerText = label;
+  checkboxWrapper.append(checkboxOption);
+  checkboxWrapper.append(checkboxLabel);
+  return checkboxWrapper;
+}
+
+html.SelectElement _getDropdown(
+    List<String> options, String selectedOption, Function(String) onChange) {
+  var dropdownSelect = html.SelectElement()
+    ..classes = ['form-control']
+    ..onChange.listen((e) => onChange((e.target as html.SelectElement).value));
+
+  for (var option in options) {
+    var dropdownOption = html.OptionElement()
+      ..setAttribute('value', option)
```

Comment:
I think you can access the `value` property directly?
```suggestion
      ..value = option
```

Reply:
Ah yes, makes sense!